### PR TITLE
compile contracts in build.rs and remove compiled from git so they do…

### PR DESCRIPTION
**Pull request by snd**
…n't get out of sync

removed compiled contracts from git (#78 ) since it became annoying and was easily forgotten having to compile the contracts manually all the time and having to commit changes to `.bin`s and `.abi`s. now the `bridge` crate always builds and uses the newest contracts. nothing can get out of sync accidentally.
the only downside is that `solc` must be installed.

### CODE REVIEW

File: bridge/build.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use std::process::Command;
+
+fn main() {
+    // rerun build script if bridge contract has changed.
+    // without this cargo doesn't since the bridge contract
+    // is outside the crate directories
+    println!("cargo:rerun-if-changed=../contracts/bridge.sol");
+    let exit_status = Command::new("solc")
```
Comment (debris) [inline bridge/build.rs]: Could we make this optional? So if there is no `solc` installed, precompiled code is used.

File: bridge/build.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        .arg("--abi")
+        .arg("--bin")
+        .arg("--output-dir").arg("../compiled_contracts")
+        .arg("--overwrite")
+        .arg("../contracts/bridge.sol")
```
Comment (debris) [inline bridge/build.rs]: should we also add argument to optimize compiled code?

Comment (snd) [reply bridge/build.rs]: definitely

File: bridge/build.rs (hunk 1 above)
Comment (snd) [reply bridge/build.rs]: we could. but then we'd be back to having to commit compiled code with people forgetting to update it etc.

File: bridge/build.rs (hunk 2 above)
Comment (snd) [reply bridge/build.rs]: done

**Comment by snd** [comment]
would be great to get this merged. it will prevent all classes of problems arising from contracts and compiled contracts getting out of sync.

if having to install `solc` before building the bridge becomes a problem in itself we can look into other solutions.

**Pull request merged**